src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -241,9 +241,17 @@ public ImageProcessor(ImageClassificationTransformer transformer)
             public Tensor ProcessImage(in VBuffer<byte> imageBuffer)
             {
                 var imageTensor = EncodeByteAsString(imageBuffer);
-                var processedTensor = _imagePreprocessingRunner.AddInput(imageTensor, 0).Run()[0];
-                imageTensor.Dispose();
-                return processedTensor;
+                try
+                {
+                    var processedTensor = _imagePreprocessingRunner.AddInput(imageTensor, 0).Run()[0];
+                    imageTensor.Dispose();
+                    return processedTensor;
+                }
+                catch (TensorflowException)
+                {
+                    imageTensor.Dispose();
```

Comment:
(minor) you can put the `Dispose()` call in a `finally` block, and it will always get disposed, even if another type of exception is thrown.

Reply:
Even better would be to use a `using` block.